packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -105,28 +102,97 @@ class BlockListBlock extends Component {
 		return blockName;
 	}
 
+	applySelectedBlockStyle() {
+		const {
+			hasChildren,
+			getStylesFromColorScheme,
+			isSmallScreen,
+			selectionIsNested,
+			isInnerBlockHolder,
+		} = this.props;
+
+		const fullSolidBorderStyle = {
+			...styles.fullSolidBordered,
+			...getStylesFromColorScheme( styles.solidBorderColor, styles.solidBorderColorDark ),
+		};
+
+		if ( hasChildren ) {
+			return { ...styles.selectedParent,	...fullSolidBorderStyle	};
+		}
+
+		if ( isSmallScreen && ! selectionIsNested && ! isInnerBlockHolder ) {
```

Comment:
To help clarity. we could group those conditions into an intermediate one whose name could explain what this is about (or simply add a comment)

Reply:
done ! I have also put inline comment inside the styling logic